addons/binding/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/handler/MystromBridgeHandler.java

```diff
@@ -0,0 +1,290 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mystrom.handler;
+
+import static org.openhab.binding.mystrom.MystromBindingConstants.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.mystrom.config.MystromBridgeConfiguration;
+import org.openhab.binding.mystrom.discovery.MystromDiscoveryService;
+import org.openhab.binding.mystrom.internal.AuthResponse;
+import org.openhab.binding.mystrom.internal.Device;
+import org.openhab.binding.mystrom.internal.MystromDeviceAddedListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * This bridge handler connects to mystrom and handles all the api requests. It pulls down the
+ * updated data, polls the system and does all the co-ordination with the other handlers
+ * to get the data updated to the correct things.
+ *
+ * @author Stéphane Raemy - initial contribution
+ */
+public class MystromBridgeHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(MystromBridgeHandler.class);
+
+    // REST Client API variables
+    protected final Client mystromClient = ClientBuilder.newClient();
```

Comment:
any reason these fields are protected and public? they don't seem to used externally.